node/forest_libp2p/src/service.rs

```diff
@@ -276,6 +276,13 @@ where
         self.network_receiver_out.clone()
     }
 }
+async fn emit_event(sender: &Sender<NetworkEvent>, event: NetworkEvent) {
+    if !sender.is_full() {
+        sender.send(event).await
+    } else {
+        warn!("network sender channel was full, ignoring event");
```

Comment:
Oof, is this okay? Is there anyway to avoid this? How often do you see us hitting his with the channel size increased to 50? I see this potentially being a problem when we have like a bajillion peers on a production network

Reply:
yeah, I can add a TODO to revisit (or maybe what can be done is keeping a vector buffer of events to send out once the channel empties). The issue with this is that if this channel fills and this send awaits, then the libp2p thread will be blocked and the whole node stalls. 

For now I think ignoring an event is fine (I haven't hit this warning yet with the current code, but let me know if you do).

Reply:
Just afraid of when we hit production systems. For reference, on space race, lotus takes 2-4 seconds to process a block. But should be ok for now

Reply:
But the thread that polls this channel is not blocked by anything, that's the point of this PR. All of that logic will be done asyncronously. (This is why even the tipset retrieval is done in a separate task, because otherwise the channel fills and stalls)